scripts/build_test.py

```diff
@@ -721,4 +723,169 @@ def mock_require_compiled_js_dir_to_be_valid():
                 self.assertFalse(
                     os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)))
 
+    def test_generate_app_yaml(self):
+        temp_file = tempfile.NamedTemporaryFile()
+        temp_file.name = 'app_dev.yaml'
+        with open('app_dev.yaml', 'w') as tmp:
+            tmp.write('Some content in app_dev.yaml')
+
+        build.generate_app_yaml()
+        with open('app.yaml', 'r') as yaml_file:
+            content = yaml_file.read()
+
+        temp_file.close()
+        checkout_cmd = 'git checkout -- app_dev.yaml'.split()
+        subprocess.call(checkout_cmd)
+        self.assertEqual(
+            content,
+            '# THIS FILE IS AUTOGENERATED, DO NOT MODIFY\n'
+            'Some content in app_dev.yaml')
+
+    def test_safe_delete_file(self):
+        temp_file = tempfile.NamedTemporaryFile()
+        temp_file.name = 'some_file.txt'
+        with open('some_file.txt', 'w') as tmp:
+            tmp.write('Some content.')
+        self.assertTrue(os.path.isfile('some_file.txt'))
+
+        build.safe_delete_file('some_file.txt')
+        self.assertFalse(os.path.isfile('some_file.txt'))
+
+    def test_minify_third_party_libs(self):
+
+        def _mock_safe_delete_file(unused_filepath):
+            """Mocks build.safe_delete_file()."""
+            pass
+
+        self.assertFalse(os.path.isfile(
+            'core/tests/data/third_party/css/third_party.min.css'))
+        self.assertFalse(os.path.isfile(
+            'core/tests/data/third_party/js/third_party.min.js'))
+        self.assertFalse(os.path.isfile(
+            'core/tests/data/third_party/js/third_party.min.js.map'))
+
+        with self.swap(build, 'safe_delete_file', _mock_safe_delete_file):
+            build.minify_third_party_libs('core/tests/data/third_party')
+
+        self.assertTrue(os.path.isfile(
+            'core/tests/data/third_party/css/third_party.min.css'))
+        self.assertTrue(os.path.isfile(
+            'core/tests/data/third_party/js/third_party.min.js'))
+        self.assertTrue(os.path.isfile(
+            'core/tests/data/third_party/js/third_party.min.js.map'))
```

Comment:
Perhaps check that their file size is smaller than the originals.

Reply:
Done